### PR TITLE
Update binaries by name

### PR DESCRIPTION
**Pull request by ernesto27**
Hi, marcos.
This is a pull requestes related to the issue https://github.com/marcosnils/bin/issues/43

### CODE REVIEW

File: cmd/update.go [hunk 1]
```diff
@@ -55,8 +56,14 @@ func newUpdateCmd() *updateCmd {
 
 			// Update single binary
 			if bin != "" {
+				path, errBin := exec.LookPath(bin)
```
Comment (marcosnils) [inline cmd/update.go:59]: I think we can be a bit smarter here and avoid the lookup if the user already sends us the path right?  If I do `bin update $(which kind)` like I do today, seems like the lookup is unnecessary. WDYT?

File: cmd/update.go [hunk 2]
```diff
@@ -138,6 +145,10 @@ func newUpdateCmd() *updateCmd {
 	return root
 }
 
+func printBinaryNotFound(bin string) error {
```
Comment (marcosnils) [inline cmd/update.go:148]: this function doesn't print anything actually. Maybe rename it to "pathBinaryNotFoundError ?

File: cmd/update.go (hunk 1 above)
Comment (ernesto27) [reply cmd/update.go:59]: Yes, maybe i would add a regexp check for the value / in the path value before call the LookPath function.

But on the other hand i looked at the LookPath docs and i found the is a check of the filestring
"If file contains a slash, it is tried directly and the PATH is not consulted." 
https://golang.org/src/os/exec/lp_unix.go?s=923:965#L24

I'm not sure if this validation of the LookPath function would be enough WDYT?

File: cmd/update.go (hunk 2 above)
Comment (ernesto27) [reply cmd/update.go:148]: Yes,  that is true ,  no problem with this.

File: cmd/update.go (hunk 1 above)
Comment (marcosnils) [reply cmd/update.go:59]: > I'm not sure if this validation of the LookPath function would be enough WDYT?

Yes, I saw that as well, but regardless if the argument contains a "/" or not, it still calls the `Stat` syscall afterwards to check if the file exists (https://go.googlesource.com/go/+/go1.15.8/src/os/exec/lp_unix.go#20). I guess this is premature optimization and it's not really necessary. Let's leave it like this for the moment, I don't think we need to change it

**Comment by ernesto27** [comment]
I added a basic check to prevent call LookPath if command is called in this way $(which command),  and change the function name.

I see that other option to  solve this issue would be to use the file HOME/.bin/config.json to check 
the path and prevent to call function LookPath, it would be a better option, WDYT?

**Review by marcosnils — APPROVED** [review]

**Pull request merged**